### PR TITLE
General: Bump minimum PHP version to 7.2

### PR DESCRIPTION
**Pull request by tbradsha**
This bumps the minimum PHP version for our plugins to 7.2, matching the earliest version of WP we support (6.6).

This is listed as a major change.

The main changes are as follows:

* Removed PHP 7.0 linting
* Removed a now-unneeded version check in `.phan/config.base.php`
* Bumped PHPCS configs and docs
* Updated PHP versions in `composer.json` and `readme.txt`
* Changed PHPUnit's `setMethods()` to `onlyMethods()` and `addMethods()` (note that the two instances of a `null` param were switched to an empty array)

I also ran across a "7.2" false positive referencing an old Gutenberg version (4.5 years old) in `projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/page-patterns-plugin.tsx` and removed the ternary.

To address in a separate PR:
* `codesniffer` updates
* other `composer` dependencies
* a deeper look at various todos

Notes:
* `addMethods()` is officially deprecated as of PHPUnit 10 (sebastianbergmann/phpunit#5320), but that'll have to wait for a future day.
* ~A test that installed an old theme (https://wordpress.org/themes/itek/) was failing. I updated it to a new theme (https://wordpress.org/themes/dossier/) and it passed. Turns out it was coincidence, and tests were intermittently failing due to heavy WP 6.7 update traffic, but I'll keep the change, as the old theme was last updated 8.5 years ago and is bound to eventually fail.~ It turns out there's something more to just changing the theme. Changed back to iTek for now.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Verify changes are valid and that CI checks are happy.

### CODE REVIEW

**Comment by tbradsha** [comment]
Oops, forgot to rebase before pushing. That's now done.

File: docs/monorepo.md [hunk 1]
```diff
@@ -266,15 +266,15 @@ If a project contains PHP tests (typically PHPUnit), it must define `.scripts.te
...
+Tests are run with a variety of supported PHP versions from 7.2 to 8.3. If you have tests that only need to be run once, run them when `PHP_VERSION` matches that in `.github/versions.sh`.
 
 #### PHP tests for non-plugins
 
 For all project types other than WordPress plugins, the necessary version of PHPUnit and/or any other tools should be pulled in via Composer.
 
 We currently make use of the following packages in testing; it's encouraged to use these rather than introducing other tools that serve the same purpose.
 
-* [yoast/phpunit-polyfills](https://packagist.org/packages/yoast/phpunit-polyfills) supplies polyfills for compatibility with PHPUnit 6.5 to 9.0, to support PHP 7.0 to 8.3.
+* [yoast/phpunit-polyfills](https://packagist.org/packages/yoast/phpunit-polyfills) supplies polyfills for compatibility with PHPUnit 7.0 to 9.0, to support PHP 7.2 to 8.3.
```
Comment (anomiex) [inline docs/monorepo.md]:
```suggestion
* [yoast/phpunit-polyfills](https://packagist.org/packages/yoast/phpunit-polyfills) supplies polyfills for compatibility with PHPUnit 8.5 to 9.6, to support PHP 7.2 to 8.3.
```

File: projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/page-patterns-plugin.tsx [hunk 2]
```diff
@@ -43,9 +43,7 @@ export function PagePatternsPlugin( props: PagePatternsPluginProps ) {
 			isWelcomeGuideActive: (
 				select( 'core/edit-post' ) as CoreEditPostPlaceholder
 			 ).isFeatureActive( 'welcomeGuide' ) as boolean, // Gutenberg 7.2.0 or higher
```
Comment (anomiex) [inline projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/page-patterns-plugin.tsx]: Does this comment still make sense?
```suggestion
			 ).isFeatureActive( 'welcomeGuide' ) as boolean,
```

File: projects/plugins/beta/changelog/update-bump_min_php_to_7.2
Comment (anomiex) [inline projects/plugins/beta/changelog/update-bump_min_php_to_7.2:1]: I see no changes in this plugin?

File: projects/plugins/mu-wpcom-plugin/changelog/update-bump_min_php_to_7.2
Comment (anomiex) [inline projects/plugins/mu-wpcom-plugin/changelog/update-bump_min_php_to_7.2:1]: No changes in this plugin either.

**Comment by anomiex** [comment]
I see a few of the Phan issues are because bumping to 7.2 makes it stop issuing `PhanParamSignatureRealMismatchHasNoParamType` for contravariant parameter types.

OTOH, some of those might be able to be adjusted to match now that we no longer have to support older versions of Symfony.

**Comment by anomiex** [comment]
Also, you may as well remove this phpcs rule exclusion while you're enabling the corresponding Phan rule.
https://github.com/Automattic/jetpack/blob/ef4ca77b5a1fd9537caf193068a7c6df18f6459c/.github/files/phpcompatibility-dev-phpcs.xml#L12-L13
It doesn't mention 7.0, but that's what it's there for.

File: docs/monorepo.md (hunk 1 above)
Comment (tbradsha) [reply docs/monorepo.md]: I was going to go back and verify the actual bump, but thanks for doing it for me!

File: projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/page-patterns-plugin.tsx (hunk 2 above)
Comment (tbradsha) [reply projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/page-patterns-plugin.tsx]: I considered removing it, but didn't have high confidence as to whether it'd be helpful or not. Likely it's not helpful.

File: projects/plugins/beta/changelog/update-bump_min_php_to_7.2
Comment (tbradsha) [reply projects/plugins/beta/changelog/update-bump_min_php_to_7.2:1]: Not sure why changelogger added these (I used `jetpack changelogger add` in the monorepo root). 😕 Maybe due to the fact I hadn't rebased on `trunk` yet. Removed now.

File: projects/plugins/mu-wpcom-plugin/changelog/update-bump_min_php_to_7.2
Comment (tbradsha) [reply projects/plugins/mu-wpcom-plugin/changelog/update-bump_min_php_to_7.2:1]: Removed.

**Comment by anomiex** [comment]
Missed one: https://github.com/Automattic/jetpack/blob/a70da82bd9db4c6e07fb1db236c5f451d652024f/projects/packages/backup/composer.json#L7

**Review by anomiex — APPROVED** [review]
Let's do it!

**Pull request merged**